### PR TITLE
Feat/friendtech key holder airdrop

### PR DESCRIPTION
**Pull request by pop-punk**
Add support for easy friend tech key holder airdrops

### CODE REVIEW

File: components/providers/FriendtechProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  const { holders } = useFriendtechData(tokenAddress);
+
+//   const parsedRecipients = useMemo(() => {
+//     try {
+//       return (
+//         recipients.length
+//           ? recipientsParser(tokenDecimals).parse(recipients)
+//           : []
+//       ) as AirdropRecipient[];
+//     } catch (e) {
```
Comment (backseats) [inline components/providers/FriendtechProvider.tsx:55]: get rid of all this commented out code?

File: components/types/modals.ts [hunk 2]
```diff
@@ -1 +1 @@
-export type ModalSelector = 'confirm' | 'congrats';
+export type ModalSelector = 'confirm' | 'congrats' | '';
```
Comment (backseats) [inline components/types/modals.ts:1]: why `| ''`?

File: components/ui/MagicTextArea.tsx [hunk 3]
```diff
@@ -75,24 +99,31 @@ const MagicTextArea: FC<Props> = (props: Props) => {
   };
 
   const handleERC20 = (value: string) => {
-    const regex = /^0x[a-fA-F0-9]{40}(?= ?[^ ])([=,]?) *(\d+(\.\d+)?)$/;
+    const regex = /^0x[a-fA-F0-9]{40}(?= ?[^ ])([=,]?) *(\d*(\.\d*)?)$/;
```
Comment (backseats) [inline components/ui/MagicTextArea.tsx:102]: Why this change?

File: components/ui/MagicTextArea.tsx [hunk 4]
```diff
@@ -122,18 +155,36 @@ const MagicTextArea: FC<Props> = (props: Props) => {
...
+          />
+        </>
+      ) : (
+        <p className="text">{instructions}</p>
+      )}
 
       <textarea
         value={textareaValue}
-        className="w-full min-h-[300px] max-h-[600px] mt-4 p-3 text-black border-black border-2 rounded-md"
+        className="w-full min-h-[200px] max-h-[400px] mt-4 p-3 text-black border-black border-2 rounded-md" // adjusted the heights
```
Comment (backseats) [inline components/ui/MagicTextArea.tsx:179]: can prob remove this comment

File: components/types/modals.ts (hunk 2 above)
Comment (kryptopaul) [reply components/types/modals.ts:1]: Previously a certain line would be be ` const [openModal, setOpenModal] = useState<ModalSelector | false>(false);`. With an empty string option within `ModalSelector` it's simpler. (`const [openModal, setOpenModal] = useState<ModalSelector>("");`)

But I've just noticed that I've only edited it within `FriendtechProvider.tsx` and it's in the old way in ETH, ERC20 and ERC721 Providers. Should I revert that change, or introduce it in the remaining providers for consistency?

File: components/ui/MagicTextArea.tsx (hunk 3 above)
Comment (kryptopaul) [reply components/ui/MagicTextArea.tsx:102]: According to @smitch88, "I put up a PR @PopPunkOnChain I also changed the regex so it takes .1 not just 0.*"

File: components/types/modals.ts (hunk 2 above)
Comment (backseats) [reply components/types/modals.ts:1]: i like an empty string over `false` as the default state so we don't mix types
